### PR TITLE
X509_cmp(): Fix comparison in case x509v3_cache_extensions() failed to due to invalid cert, backport to 1.1.1

### PR DESCRIPTION
**Pull request by DDvO**
This backports the main part of #13755, namely the fix for #13698 as discussed there,
by checking more specifically if the hash value computation succeeded.
This required improving the error reporting of `x509v3_cache_extensions()`.

Adds a test using the example cert given there, which has invalid SKID and AKID encodings.

- [x] tests are added or updated

### CODE REVIEW

File: doc/man3/X509_get_extension_flags.pod [hunk 1]
```diff
@@ -78,12 +78,17 @@ The certificate contains an unhandled critical extension.
...
-certificate should be rejected.
+Some certificate extension values are invalid or inconsistent.
+The certificate should be rejected.
 This bit may also be raised after an out-of-memory error while
 processing the X509 object, so it may not be related to the processed
 ASN1 object itself.
 
+=item B<EXFLAG_NO_FINGERPRINT>
+
+Failed to compute the internal SHA1 hash value of the certificiate.
```
Comment (misery) [inline doc/man3/X509_get_extension_flags.pod:89]: typo: certificiate

Comment (DDvO) [reply doc/man3/X509_get_extension_flags.pod]: Thx, fixed.

**Comment by DDvO** [comment]
Since the upstream fix #13755 is meanwhile approved and merged
and the relevant adaptations have been taken over here, 
this is now due for review.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @t8m

**Pull request closed without merging**